applications/chain-extension-suite.md

```diff
@@ -0,0 +1,240 @@
+
+
+## Development Status :open_book:
+
+### Milestone 1
+
+- **Estimated Duration:** 2 month
+- **FTE:** 3
+- **Costs:** $35,000
+
+| Number | Deliverable          | Specification |
+| ------ | -------------------- | ------------- |
+| **0a.**| License              | Apache 2.0    |
+| **0b.**| Documentation        | Comprehensive documentation for each pallet extension and the upgrade, including tutorials and guides on how to integrate and utilize the extensions. |
+| **0c.**| Testing Guide        | A detailed guide on how to run the test suites for each extension and the upgraded pallet_hookpoints, along with a demonstration of the testing coverage. |
+| **1.** | `pallet_nft_extensions` | Development and implementation of the `pallet_nft_extensions`, providing chain extension functionality for `pallet_nft` in both ink! and EVM ecosystems. |
+| **2.** | `pallet_assets_extension` | Development and implementation of the `pallet_assets_extension`, providing chain extension functionality for `pallet_assets` in both ink! and EVM ecosystems. |
+| **3.** | `pallet_hookpoints` v5 extension | Upgrading `pallet_hookpoints` to be compatible with ink! v5, ensuring its functionality aligns with the latest advancements in ink!. |
+| **4.** | Medium Article | A Medium article detailing and announcing the new pallet extensions and the upgrade of `pallet_hookpoints` to ink! v5. |
+
+---
+
+### Milestone 2
+
+- **Estimated Duration:** 3-4 month
+- **FTE:**  3
+- **Costs:** $65,000
+
+| Number | Deliverable          | Specification |
+| ------ | -------------------- | ------------- |
+| **0a.**| License              | Apache 2.0    |
+| **0b.**| Documentation        | Comprehensive documentation covering the CLI Extension tool, its features, and how to use it for generating chain extensions. |
+| **0c.**| Testing Guide        | A detailed guide on how to run the test suites for the CLI Extension tool and a demonstration of the testing coverage. |
+| **1.** | CLI Extension Tool - Part 1: Setup   | Establishing the CLI tool to inspect the Substrate runtime and enable code generation capabilities. |
+| **2.** | CLI Extension Tool - Part 2: Substrate-side Chain Extensions Generation | Auto-generation of the Substrate-side chain extension code based on the inspected runtime methods. |
+| **3.** | CLI Extension Tool - Part 3: ink!-side Chain Extension Traits Generation | Auto-generation of ink!-side chain extension traits to facilitate communication with the generated Substrate-side chain extensions. |
+| **4.** | CLI Extension Tool - Part 4: Reference Smart Contract Generation | Generation of a reference smart contract in ink! that interacts with the generated chain extension traits and Substrate-side chain extensions. |
+| **5.** | Medium Article | A Medium article detailing and announcing the CLI Extension tool, explaining how it simplifies the generation of chain extensions and enhances the developer experience within the Substrate ecosystem. |
```

Comment:
My thoughts here:

- Will there be any limitations on which pallets are supported by the CLI? Can they be used for any pallet, for any attribute/function/trait/etc.?
- I think you can skip the _CLI Extension Tool - Part 1: Setup_ delivery, as this essentially only describe a CLI tool with very unclearly scoped "code generation capabilities". In any way, the further 3 deliverables cover the same CLI tool.

Reply:
> Will there be any limitations on which pallets are supported by the CLI? Can they be used for any pallet, for any attribute/function/trait/etc.?

The idea is to work with the introspection that is e.g. used by polkadot.js.org and autogenerate contracts based on this. This may prove to be to complicated for some extrinsic, but we should be able to provide boilerplate code that eases development work, which is our main goal.

Reply:
>The idea is to work with the introspection that is e.g. used by polkadot.js.org and autogenerate contracts based on this. This may prove to be to complicated for some extrinsic, but we should be able to provide boilerplate code that eases development work, which is our main goal.

So in other words, the CLI would be usable for any extrinsic in any pallet, except for some unexpected edge cases that might arise, right?